experimental/dds/tree/src/migration-shim/migrationDeltaHandler.ts

```diff
@@ -100,6 +101,13 @@ export class MigrationShimDeltaHandler implements IShimDeltaHandler {
 		return this.treeDeltaHandler.process(message, local, localOpMetadata);
 	}
 
+	public processMessages(messagesCollection: IRuntimeMessageCollection): void {
```

Comment:
I was looking at the definition of IRuntimeMessageCollection, and noticed is should probably be `@sealed`, have readonly fields, and have the messagesContent array be readonly as well, since  I don't think we intend for anything we pass them to do modify them, nor do we want third parties to implement this interface themselves.

If the interface were sealed, then it would be a non-breaking change to add a `.map` method over it which takes a "process" callback so we could avoid duplicating this logic in multiple places. I'm not sure that would be worth doing, but its an example of the additional choices you get to have if something is sealed.

You can also always remove sealing, but you can't add it (since its breaking to do so)

Since I think this interface was added since the last release, you may be able to modify it now without issue.

Reply:
Adding a mapRuntimeMessageCollection free function somewhere (likely as internal) would be another option for avoiding this duplication.

Reply:
Good suggestion. Changed it to sealed and made the properties readonly.